components/brave_perf_predictor/browser/bandwidth_savings_predictor.h

```diff
@@ -0,0 +1,43 @@
+/* Copyright (c) 2019 The Brave Authors. All rights reserved.
+ * This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this file,
+ * You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+#ifndef BRAVE_COMPONENTS_BRAVE_PERF_PREDICTOR_BROWSER_BANDWIDTH_SAVINGS_PREDICTOR_H_
+#define BRAVE_COMPONENTS_BRAVE_PERF_PREDICTOR_BROWSER_BANDWIDTH_SAVINGS_PREDICTOR_H_
+
+#include <string>
+
+#include "base/containers/flat_map.h"
+#include "components/page_load_metrics/common/page_load_metrics.mojom.h"
+#include "content/public/common/resource_load_info.mojom.h"
+
+namespace brave_perf_predictor {
+
+class BandwidthSavingsPredictor {
```

Comment:
I'd add some comments saying that this class gathers some info during the navigation and then can be queried for a certain value. `|Reset|` should be called in between of navigations, etc.